fractal_server/app/api/v1/workflow.py

```diff
@@ -107,33 +108,86 @@ async def get_workflow(
     return workflow
 
 
-@router.post("{_id}/add-task/", response_model=WorkflowRead)
+@router.post(
+    "/{_id}/add-task/",
+    response_model=WorkflowRead,
+    status_code=status.HTTP_201_CREATED,
+)
 async def add_task_to_workflow(
     _id: int,
     new_task: WorkflowTaskCreate,
     user: User = Depends(current_active_user),
     db: AsyncSession = Depends(get_db),
 ):
-    raise NotImplementedError
+    # TODO move check autorization as first thing (issue #171)
+    workflow = await db.get(Workflow, _id)
+    if not workflow:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="Workflow not found"
+        )
+    # check autorization
+    await get_project_check_owner(
+        project_id=workflow.project_id,
+        user_id=user.id,
+        db=db,
+    )
+    await workflow.insert_task(new_task.task_id, db=db)
+
+    db.merge(workflow)
+    db.commit()
+    await db.refresh(workflow)
+
+    return workflow
 
 
 @router.delete(
-    "{_id}/rm-task/{task_id}", status_code=status.HTTP_204_NO_CONTENT
+    "/{_id}/rm-task/{task_id}/", status_code=status.HTTP_204_NO_CONTENT
```

Comment:
see above.
let's call it `workflow_task_id` instead of `task_id`, to make it clear that we are selecting the task from the task_list by primary key.